checks.d/burrow.py

```diff
@@ -132,30 +136,32 @@ def _consumer_groups_offsets(self, clusters, burrow_address, extra_tags):
             consumers_path = "%s/%s/consumer" % (CLUSTER_ENDPOINT, cluster)
             consumers_list = self._rest_request_to_json(burrow_address, consumers_path).get("consumers", [])
             for consumer in consumers_list:
-                topics_path = "%s/%s/topic" % (consumers_path, consumer)
-                topics_list = self._rest_request_to_json(burrow_address, topics_path).get("topics", [])
-                for topic in topics_list:
-                    topic_path = "%s/%s" % (topics_path, topic)
-                    response = self._rest_request_to_json(burrow_address, topic_path)
-                    if not response:
-                        continue
+                topics_path = "%s/%s" % (consumers_path, consumer)
+                topics = self._rest_request_to_json(burrow_address, topics_path).get("topics", [])
+                for topic in topics:
                     tags = ["topic:%s" % topic, "cluster:%s" % cluster,
                             "consumer:%s" % consumer] + extra_tags
-                    self._submit_offsets_from_json(offsets_type="consumer", json=response, tags=tags)
+                    self._submit_offsets_from_json(offsets_type="consumer", offsets_list=topics[topic], tags=tags)
 
-    def _submit_offsets_from_json(self, offsets_type, json, tags):
+    def _submit_offsets_from_json(self, offsets_type, offsets_list, tags):
         """
         Find the offsets and push them into the metrics
         """
-        offsets = json.get("offsets")
-        if offsets:
-            # for unconsumed or empty partitions, change an offset of -1 to 0 so the
-            # sum isn't affected by the number of empty partitions.
-            offsets = [max(offset, 0) for offset in offsets]
-            self.gauge("kafka.%s.offsets.total" % offsets_type, sum(offsets), tags=tags)
-            for partition_number, offset in enumerate(offsets):
-                new_tags = tags + ["partition:%s" % partition_number]
-                self.gauge("kafka.%s.offsets" % offsets_type, offset, tags=new_tags)
+        # for unconsumed or empty partitions, change an offset of -1 to 0 so the
+        # sum isn't affected by the number of empty partitions.
+        offsets = []
+        for partition_number, offset_info in enumerate(offsets_list):
+            new_tags = tags + ["partition:%s" % partition_number]
+            latest_offset = None
+            offset_timestamp = 0
+            for offset in offset_info["offsets"]:
+                if not offset:
+                    continue
+                if offset["timestamp"] > offset_timestamp:
+                    latest_offset = offset["offset"]
+            self.gauge("kafka.%s.offsets" % offsets_type, latest_offset, tags=new_tags)
+        offsets = [max(offset, 0) for offset in offsets]
```

Comment:
`offsets` is an empty list in #152, and doesn't appear to change at all before here, where an empty list is iterated over